src/client/pydaos/pydaos_core.py

```diff
@@ -69,20 +75,25 @@ class DCont():
         Create new DArray object.
     """
 
-    def __init__(self, pool=None, cont=None, path=None):
+    def __init__(self, pool=None, cont=None, path=None, open_mode='RW'):
         self._dc = DaosClient()
         self._hdl = None
         if path is None and (pool is None or cont is None):
             raise PyDError("invalid pool or container UUID",
                            -pydaos_shim.DER_INVAL)
+        open_ro = False
+        if open_mode == 'RO':
+            open_ro = True
+        elif open_mode != 'RW':
+            raise InvalidModeError('open_mode is neither "RO" or "RW"')
         if path is not None:
             self.pool = None
             self.cont = None
-            (ret, hdl) = pydaos_shim.cont_open_by_path(DAOS_MAGIC, path, 0)
+            (ret, hdl) = pydaos_shim.cont_open_by_path(DAOS_MAGIC, path, open_ro)
```

Comment:
I think it's better if we can pass RW or RO by flags

Reply:
Couldn't it just be similar to this?
https://github.com/daos-stack/daos/blob/80f92739b294d97004fe5db141139030a7dc803c/src/client/pydaos/pydaos_shim.c#L1484

Maybe even just
```
PyModule_AddIntConstant(module, "DAOS_COO_RO", DAOS_COO_RO);
PyModule_AddIntConstant(module, "DAOS_COO_RW", DAOS_COO_RW);
PyModule_AddIntConstant(module, "DAOS_COO_EX", DAOS_COO_EX);
...
```

Reply:
Are you proposing a change to the (external) python API here or the (internal) C API?

Reply:
It would be a change in `pydaos_shim.c` to export definitions based on the core C code.
E.g. it's used like this
https://github.com/daos-stack/daos/blob/80f92739b294d97004fe5db141139030a7dc803c/src/client/pydaos/pydaos_core.py#L86


So in `pydaos_shim.c` we'd have
```
PyModule_AddIntConstant(module, "DAOS_COO_RO", DAOS_COO_RO);
```

and in the python we'd do
```
pydaos_shim.DAOS_COO_RO
```

Reply:
Going forward that would be better I agree but whilst there's only one value I don't see the benefit in making that change yet.  This is all internal so there's no compatibility issue that requires us to commit to something now.